src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.cs

```diff
@@ -329,15 +329,10 @@ public override string ToString()
 
         public override MethodInfo GetBaseDefinition() => this;
 
-        public override ParameterInfo[] GetParameters()
-        {
-            ParameterInfo[] privateParameters = LoadParameters();
-            ParameterInfo[] parameters = new ParameterInfo[privateParameters.Length];
-            Array.Copy(privateParameters, parameters, privateParameters.Length);
-            return parameters;
-        }
+        public override ParameterInfo[] GetParameters() =>
+            GetParametersAsSpan().ToArray();
 
-        internal override ParameterInfo[] GetParametersNoCopy() => LoadParameters();
+        internal override ReadOnlySpan<ParameterInfo> GetParametersAsSpan() => LoadParameters();
```

Comment:
You could move LoadParameters implementation here

Reply:
I could, but it would complicate DefineParameter, which uses members defined on the derived RuntimeParameterInfos it gets back from LoadParameters.